models/repo/repo_unit.go

```diff
@@ -202,6 +202,33 @@ func (cfg *ActionsConfig) ToDB() ([]byte, error) {
 	return json.Marshal(cfg)
 }
 
+// ProjectsMode represents the projects enabled for a repository
+type ProjectsMode string
+
+const (
+	// ProjectsModeRepo allows only repo-level projects
+	ProjectsModeRepo ProjectsMode = "repo"
+	// ProjectsModeOwner allows only owner-level projects
+	ProjectsModeOwner ProjectsMode = "owner"
+	// ProjectsModeAll allows both kinds of projects
+	ProjectsModeAll ProjectsMode = "all"
```

Comment:
What is `none`?
Globally disabled, `nil`, or something else?

Reply:
I don't exactly understand what you mean.... It makes no sense to allow a `ProjectMode = "none"` here because it is already handled by disabling the projects repo unit.

Reply:
Yes, currently.
However, I'm wondering if it wouldn't make sense to add it to this enum of options.
So that you can always use `projectUnit.Supports("repo")` for example.

Reply:
Without the current `projectUnit.Enabled && projectUnit.Supports("repo")`

Reply:
I added `ProjectsModeNone` and a the permission check now looks the following:

`projectsUnit.ProjectsConfig().IsProjectsAllowed(ProjectsModeRepo)`